### PR TITLE
Make a3m transfers configurable

### PR DESCRIPTION
**Pull request by Diogenesoftoronto**
The a3m submissions were previously hardcoded. Now defaults are set and the configuration is with the rest of a3m.
This has not been tested.

### CODE REVIEW

**Comment by sevein** [comment]
> This has not been tested.

Is this still true? Should I wait until you have a chance to test?

**Comment by scollazo** [comment]
Could you add some examples about how to use the configuration settings to enduro.toml ?

**Comment by scollazo** [comment]
The processing configuration works part works as expected, could you remove the changes regarding aip naming, and leave them as part of #656 ?

File: enduro.toml
Comment (djjuhasz) [inline enduro.toml]: I think this should be removed, as the app name doesn't need to be configurable.

File: internal/a3m/a3m.go
Comment (djjuhasz) [inline internal/a3m/a3m.go]: Remove this block too please.

File: internal/a3m/a3m.go
Comment (djjuhasz) [inline internal/a3m/a3m.go]: Revert this please.

File: internal/a3m/a3m.go
Comment (djjuhasz) [inline internal/a3m/a3m.go]: Revert

File: internal/a3m/config.go
Comment (djjuhasz) [inline internal/a3m/config.go]: Nitpick: I would label this import "transferService", a3m_transferService" or "a3mTS" - something that avoids coding the API version into the label.  If the transferservice API is updated at some point, then the label won't need to be changed to match.

**Review by djjuhasz — CHANGES_REQUESTED** [review]
@Diogenesoftoronto I flagged the occurrences of the customizable app name that I think should be removed, as per @scollazo request.  I also noted one little naming nitpick.

File: internal/a3m/config.go
Comment (sevein) [reply internal/a3m/config.go]: It's not uncommon to do it for some reason, e.g. see [gc_controller.go](https://github.com/kubernetes/kubernetes/blob/2c6c4566eff972d6c1320b5f8ad795f88c822d09/pkg/controlplane/controller/apiserverleasegc/gc_controller.go#L26) (many other examples available in that project). Since the version number relates to the `major` component (SemVer), maybe it's a way to signal that if it changes it's backward-incompatible so a new name is the best way to reflect that.
If you go with just the name I'd suggest `transferservice` since that's the original name of the package, `transferService` may be understood as an instance of something.

File: enduro.toml
Comment (sevein) [inline enduro.toml]: For consistency, should be `[a3m.processing]` (e.g. see `[api.auth]`, etc).
I don't think Viper cares.

File: internal/a3m/a3m.go
Comment (sevein) [inline internal/a3m/a3m.go]: To be clear, the zero value of a type such Processing is an instance where all its fields are set to their respective zero values, e.g. `Normalize: false`, `AipCompressionLevel: 0`, etc... but if someone purposely sets all fields to zero this block would overwrite the user's config with the defaults you're proposing, that would be misleading.

If you want to provide configuration defaults I think you'd need to [establish defaults using Viper's API](https://pkg.go.dev/github.com/spf13/viper#readme-establishing-defaults). You could try to declare a new `ProcessingDefaults` in the `a3m` package and refer to it from `internal/config/config.go` (`Read`), e.g.:

```go
v.SetDefault("a3m.processing", a3m.ProcessingDefaults)
```

File: internal/a3m/config.go
Comment (sevein) [inline internal/a3m/config.go]: It would be nice to add a comment in Processing to clarify the source of all these fields, e.g.:

    // Processing mirrors all the processing configuration fields in transferservicev1beta1.ProcessingConfig.
    // The goal is to reflect all the fields available in the upstream API.

Viper provides its own extensible [unmarshal](https://pkg.go.dev/github.com/spf13/viper#readme-unmarshaling) mechanism, maybe something we could explore in the future.

**Review by sevein — CHANGES_REQUESTED** [review]
A few suggestions.

File: internal/a3m/config.go
Comment (Diogenesoftoronto) [reply internal/a3m/config.go]: This is what the lsp suggests. I was just using that.

**Review by djjuhasz — COMMENTED** [review]
@Diogenesoftoronto looks good to me, but I think we should wait on merging it until the fix for #666 is merged and we can confirm all the tests pass.

**Comment by djjuhasz** [comment]
@Diogenesoftoronto looks good to me.  Please make sure to squash all of the commits before rebasing and merging this to main.

**Pull request merged**
